lib.ml

```diff
@@ -683,6 +683,11 @@ let env : (string * (located_primitive * Types.datatype * pure)) list = [
    datatype "(Xml) ~> mu n.[ [|Text:String | Attr:(String, String) | Node:(String, n) |] ]",
    IMPURE);
 
+  "variantToXml",
+  (`Server (p1 (fun v -> `List (List.map (fun i -> `XML (i)) (Value.xml_of_variants v)))),
+   datatype "(mu n.[ [|Text:String | Attr:(String, String) | Node:(String, n) |] ]) ~> Xml",
```

Comment:
Not a huge fan of having the attributes floating around in the XML element list, as opposed to attached to the Node (e.g. as a list of attribute/value paires, or better a finite map) but this is fine for now.
Oh... I see that this is just reflecting what Links does internally